tests/test_executor.py

```diff
@@ -66,40 +87,67 @@ def test_py_dialect(self):
         self.assertEqual(generate(parse_one("x is null")), "scope[None][x] is None")
 
     def test_optimized_tpch(self):
-        for i, (sql, optimized) in enumerate(self.sqls[:20], start=1):
+        for i, (sql, optimized) in enumerate(self.tpch_sqls[:20], start=1):
             with self.subTest(f"{i}, {sql}"):
                 a = self.cached_execute(sql)
-                b = self.conn.execute(optimized).fetchdf()
+                b = self.tpch_conn.execute(optimized).fetchdf()
                 self.rename_anonymous(b, a)
                 assert_frame_equal(a, b)
 
+    def subtestHelper(self, i, table, type_exec="tpc-h"):
+        with self.subTest(f"{type_exec} {i + 1}"):
+            if type_exec == "tpc-h":
+                sql, _ = self.tpch_sqls[i]
+            else:
+                sql, _ = self.tpcds_sqls[i]
+            a = self.cached_execute(sql, type_exec)
+            b = pd.DataFrame(
+                ((np.nan if c is None else c for c in r) for r in table.rows),
+                columns=table.columns,
+            )
+            assert_frame_equal(a, b, check_dtype=False, check_index_type=False)
+
     def test_execute_tpch(self):
         def to_csv(expression):
             if isinstance(expression, exp.Table) and expression.name not in ("revenue"):
                 return parse_one(
-                    f"READ_CSV('{DIR}{expression.name}.csv.gz', 'delimiter', '|') AS {expression.alias_or_name}"
+                    f"READ_CSV('{DIR_TPCH}{expression.name}.csv.gz', 'delimiter', '|') AS {expression.alias_or_name}"
                 )
             return expression
-
+        
         with Pool() as pool:
             for i, table in enumerate(
                 pool.starmap(
                     execute,
                     (
                         (parse_one(sql).transform(to_csv).sql(pretty=True), TPCH_SCHEMA)
-                        for sql, _ in self.sqls
+                        for sql, _ in self.tpch_sqls
+                    ),
+                )
+            ):
+                self.subtestHelper(i, table, "tpc-h")
+
+    def test_execute_tpcds(self):
```

Comment:
Looks like we can probably [dry](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) this up a bit more. You can give it a shot.